packages/eslint-plugin/src/rules/class-literal-property-style.ts

```diff
@@ -89,16 +89,28 @@ export default createRule<Options, MessageIds>({
             return;
           }
 
+          const sourceCode = getSourceCode(context);
+          const name = sourceCode.getText(node.key);
+
+          if (node.parent.type === AST_NODE_TYPES.ClassBody) {
+            const hasDuplicateKeySetter = node.parent.body.some(
+              element =>
+                element.type === AST_NODE_TYPES.MethodDefinition &&
+                element.kind === 'set' &&
+                sourceCode.getText(element.key) === name,
```

Comment:
[Bug] `getText` works nicely for comments in auto-fixers... but can be buggy when trying to compare keys declared in different ways.

```ts
class Mx {
  set p1(val) {}
  get ['p1']() {
    return '';
  }
}
```

That causes a test failure right now if added as `valid`, though IMO it shouldn't.

You should be able to use `getStaticValue(node.key)?.value ?? sourceCode.getText(node.key)` (and the same for `element`).

This is a pretty rare edge case and I wouldn't be surprised if it's not handled in other rules. If I'm forgetting a case where we decided not to support such shenanigans then please let me know, we can just ignore this comment. 😄

Reply:
@JoshuaKGoldberg I haven't checked all the rules, but it seems to support computed key in general.
I fixed. Thanks for the review.